python/tank/bootstrap/manager.py

```diff
@@ -60,22 +61,110 @@ def __repr__(self):
+        is defined via :meth:`base_configuration` will always be used.
+        """
+        return self._do_shotgun_config_lookup
+
+    def _set_do_shotgun_config_lookup(self, status):
+        """
+        Flag to indicate if the bootstrap process should connect to
+        Shotgun and attempt to resolve a config. Defaults to True.
+
+        If True, the bootstrap process will connect to Shotgun as part
+        of the startup, look for a pipeline configuration and attempt
+        to resolve a toolkit environment to bootstrap into via the
+        Pipeline configuration data. Failing this, it will fall back on
+        the :meth:`base_configuration`.
+
+        If False, no Shotgun lookup will happen. Instead, whatever config
+        is defined via :meth:`base_configuration` will always be used.
+        """
+        self._do_shotgun_config_lookup = status
+
+    do_shotgun_config_lookup = property(_get_do_shotgun_config_lookup, _set_do_shotgun_config_lookup)
+
+
+    def _get_entry_point(self):
+        """
+        The entry point defines the scope of the bootstrap operation.
+
+        If you are bootstrapping into an entire Toolkit pipeline, e.g
+        a traditional Toolkit setup, this should be left blank.
+
+        If you are writing a plugin that is intended to run side by
+        side with other plugins in your target environment, the entry
+        point will be used to define a scope and sandbox in which your
+        plugin will execute. For example, if your plugin is packaging up
+        review tools for RV, name your entry point "RV Review". If your plugin
+        embeds the shotgun panel in a Maya Plugin, name it "Maya Panel".
+
+        The entry point value can be used to customize the behavior of a
+        plugin via Shotgun. At bootstrap, toolkit will look for a pipeline
+        configuration with a matching name and entry point. If found, this
+        will be used instead of the one defined by the :meth:`base_configuration`
+        property.
+
+            .. note:: If you want to force the :meth:`base_configuration` to always
+                      be used, set :meth:`do_shotgun_config_lookup` to False.
+        """
+        return self._entry_point
+
+    def _set_entry_point(self, entry_point):
+        """
+        The entry point defines the scope of the bootstrap operation.
+
+        If you are bootstrapping into an entire Toolkit pipeline, e.g
+        a traditional Toolkit setup, this should be left blank.
+
+        If you are writing a plugin that is intended to run side by
+        side with other plugins in your target environment, the entry
+        point will be used to define a scope and sandbox in which your
+        plugin will execute. For example, if your plugin is packaging up
+        review tools for RV, name your entry point "RV Review". If your plugin
+        embeds the shotgun panel in a Maya Plugin, name it "Maya Panel".
```

Comment:
They do need to be display pretty. In Shotgun, when you override the config used by a plugin, it would be done by specifying the entry point, so it would end up looking something like this:

![image](https://cloud.githubusercontent.com/assets/337710/15934800/c2039ad2-2e5c-11e6-832f-c55a041a7437.png)

I'll add some notes about uniqueness!